client/packages/common/src/authentication/api/hooks/useLogin.ts

```diff
@@ -53,22 +53,31 @@ export const useLogin = (
   const getUserPermissions = useGetUserPermissions();
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const [_error, setError, removeError] = useLocalStorage('/auth/error');
+  const upsertMostRecentCredential = (
+    username: string,
+    store?: UserStoreNodeFragment
+  ) => {
+    const newMRU = [
+      { username, store },
+      ...(mostRecentlyUsedCredentials ?? []).filter(
+        mru => mru.username.toLowerCase() !== username.toLowerCase()
+      ),
+    ];
+    setMRUCredentials(newMRU);
+  };
 
   // returns MRU store, if set
   // or the first store in the list
   const getStore = async (userDetails?: Partial<UserNode>) => {
     const defaultStore = userDetails?.defaultStore;
     const stores = userDetails?.stores?.nodes;
+    const mru = mostRecentlyUsedCredentials?.find(
+      item =>
+        item.username.toLowerCase() === userDetails?.username?.toLowerCase()
+    );
```

Comment:
The only problem is when you switch branches without clearing local storage, this throws an error, as `.find()` is not a valid function of the previous `'/mru/credentials'` object (you've changed it to an array). This would be a problem the first time a user tries to use a version with this change in it, wouldn't it?

Perhaps just add an additional check here to ensure that `mostRecentlyUsedCredentials` is an array, and if not , set it to an empty array?

Reply:
oh.. good point! on to it

Reply:
@CarlosNZ accounted for previous object format in 5b1a1c84a
Is that ok?